Docs/BazelForiOSDevelopers.md

```diff
@@ -0,0 +1,370 @@
+
+First, let's walk through creating a basic iOS application. In the root of the
+project, let's create the `BUILD` file. First, a library for the iOS
+application sources. The following code defines an `objc_library`, `sources`.
+In Xcode this is similar to navigating in the GUI and hitting `File -> New
+Target` 
+
+```
+# /path/to/myproject/BUILD
+objc_library(
+    name="sources",
+    srcs=["main.m"]
+)
+```
+
+Next, create the application target with `rules_apple`'s `ios_applicaiton` rule.
+```
+# /path/to/myproject/BUILD
+...
+load("@build_bazel_rules_apple//apple:ios.bzl", "ios_application")
+ios_application(
+    name = "ios-app",
+    bundle_id = "com.bazel-bootcamp.some",
+    families = ["iphone"],
+    infoplists = ["Info.plist"],
+    minimum_os_version = "[MINIMUM VERSION]",
+    # Add `sources` as a dependency
+    deps = [ ":sources" ],
+)
+```
+
+The`ios_application` and `objc_library` targets together would be represented
+in Xcode as:
+
+![Docs](XcodeExampleOfiOSProject.png)
+
+### Rules
+
+In the previous segment, we created an iOS application with a single BUILD
+file. The `ios_application` rule is implemented by `rules_apple` and the
+`objc_library` is a native rule.
+
+A target is an instance of a rule. Rules implement business logic for how the
+iOS application is built by creating actions. actions represent invocations of
+extenal command line programs like `clang` or `bash`. Typical projects contain
+many rules, targets, and BUILD files.
+
+```
+BUILD file -> target -> rule -> action -> execution
+```
+
+Bazel and open source rules should provide most functionality to build an iOS
+application. Generally, defining custom rules isn't required but can improve
+and consolidate functionality. To learn more about how to create custom rules,
+the document, [Bazel extensions](https://docs.bazel.build/versions/master/skylark/concepts.html)
+contains a comprehensive overview.
+	
+### Command line usage
+
+In Xcode, command line builds are achieved through the command line interface
+`xcodebuild`.  to build the scheme `ios-app` from `MyProject.xcworkspace`,
+`xcodebuild`, would be invoked as:
+
+```
+xcodebuild -workspace MyProject.xcworkspace -scheme ios-app
+```
+
+Bazel exposes a command line which can also do builds. In Bazel, _every_ target
+is in the global WORKSPACE and there is no notion of schemes.
+```
+bazel build ios-app
+```
+
+The Bazel command line has hundreds of options, which can be found in the
+[Bazel
+documentation](https://docs.bazel.build/versions/master/command-line-reference.html).
+For iOS developers, a set of useful flags is available at
+[bazel-ios-users](https://github.com/ios-bazel-users/ios-bazel-users/blob/master/UsefulFlags.md).
+
+In addition to the default bazel options, it's common to create custom
+configuration settings to customize builds, so the possible Bazel command line
+permutations are endless.
+
+For example, the following build file has conditional `copts` on `:app_store`
```

Comment:
for a Bazel layperson, COPTS is one of those things that I can't wrap my head around. Bazel docs list what it does, but not _why_ it's named COPTS. It'd be nice to have a super short explainer here

Reply:
I agree it'd probably be useful to have a segment on C++ compiler flags which are split between:
- toolchains
- objc_library
- bazelrc
- defines